perses/tests/test_repex.py

```diff
@@ -182,11 +185,243 @@ def test_RESTCapableHybridTopologyFactory_repex_charge_mutation():
+                nonbonded_force = hybrid_system.getForce(force_dict['NonbondedForce_sterics'])
+                custom_force.setUseLongRangeCorrection(False)
+                nonbonded_force.setUseDispersionCorrection(True)
+
+                # Create a reporter
+                selection = "not water"
+                reporter_file = os.path.join(temp_dir, f"{ligand_A_index}_{ligand_B_index}_{phase}.nc")
+                reporter = MultiStateReporter(
+                    reporter_file,
+                    analysis_particle_indices=htf.hybrid_topology.select(selection),
+                    checkpoint_interval=100)
+
+                # Build the hybrid repex sampler
+                sampler = HybridRepexSampler(
+                    mcmc_moves=mcmc.LangevinDynamicsMove(
+                        timestep=4.0 * unit.femtoseconds,
+                        collision_rate=1.0 / unit.picosecond,
+                        n_steps=250,
+                        reassign_velocities=False,
+                        n_restart_attempts=20,
+                        constraint_tolerance=1e-06),
+                    replica_mixing_scheme='swap-all',
+                    hybrid_factory=htf,
+                    online_analysis_interval=None)
+
+                sampler.setup(
+                    n_states=n_states,
+                    temperature=300 * unit.kelvin,
+                    storage_file=reporter,
+                    minimisation_steps=0,
+                    endstates=True)
+
+                sampler.energy_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+                sampler.sampler_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+
+                # Run repex
+                sampler.extend(n_iterations)
+
+                # Retrieve free energy
+                reporter.close()
+                reporter = MultiStateReporter(reporter_file)
+                analyzer = MultiStateSamplerAnalyzer(reporter, max_n_iterations=n_iterations)
+                # f_ij, df_ij = analyzer.get_free_energy()
+                # Extract uncorrelated energy matrix (u_ln) and samples from states (N_l)
+                energy_matrix = analyzer._unbiased_decorrelated_u_ln
+                sampled_states = analyzer._unbiased_decorrelated_N_l
+                # Compute free energies with boostrapping using pymbar
+                mbar = pymbar.MBAR(energy_matrix, sampled_states, nbootstraps=200)
+                f_ij, df_ij = mbar.getFreeEnergyDifferences(uncertainty_method="bootstrap", return_theta=False)
+
+                data[f"{ligand_A_index}-{ligand_B_index}_{phase}"] = {'free_energy': f_ij[0, -1], 'error': df_ij[0, -1]}
+
+        forward_DG = data['0-1_complex']['free_energy'] - data['0-1_solvent']['free_energy']
+        reverse_DG = data['1-0_complex']['free_energy'] - data['1-0_solvent']['free_energy']
+        DDG = abs(forward_DG - reverse_DG)
```

Comment:
We may want to avoid using `DDG` as the variable name — this makes sense for the tests with 2 legs but is may be misleading for tests with 4 legs — this is my mistake.

Reply:
and we'll want to remember to fix the sign here: ` DDG = abs(forward_DG - reverse_DG)`

Reply:
Should be already done in https://github.com/choderalab/perses/pull/1065/commits/89771675edd04414c4d9d82edbc1efe1e7624f7f